Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java

```diff
@@ -293,38 +301,53 @@ public ITransformedNetwork getTransformedNetworkusingNetworkList(List<INetwork>
      * @return ITransformedNetwork
      */
     private ITransformedNetwork getFinalTransformedNetwork(ITransformedNetwork transformedNetwork,
-            List<String> alternativeIdsForConcept) {
+            List<List<String>> alternativeIdsForConceptList) {
         // Select the nodes with the concept id that is present in the list of
         // alternative id for a concept.
         // Add all the statement id of the selected node to a set.
         // The alternative id list if not empty, is sure to contain the concept
         // id corresponding to the searched concept.
-        Set<String> statementIdSearchSet = new HashSet<String>();
+        List<Set<String>> statementIdSearchSetList = new ArrayList<Set<String>>();
+        Set<String> statementIdSet = new HashSet<String>();
         List<Node> searchedNodes = new ArrayList<Node>();
+
+        for (int i = 0; i < alternativeIdsForConceptList.size(); i++) {
+            statementIdSearchSetList.add(new HashSet<String>());
+        }
+
         for (Node node : transformedNetwork.getNodes().values()) {
             // Check if the node's concept id is present in the alternative id
             // list of the concept.
-            for (String alternativeId : alternativeIdsForConcept) {
-                if (alternativeId.equals(node.getConceptId())) {
+            int index = 0;
+            for (List<String> alternativeIdsForConcept : alternativeIdsForConceptList) {
+                if (alternativeIdsForConcept.contains(node.getConceptId())) {
                     searchedNodes.add(node);
-                    statementIdSearchSet.addAll(node.getStatementIds());
+                    statementIdSearchSetList.get(index).addAll(node.getStatementIds());
                     break;
                 }
+                index += 1;
+            }
+        }
+        // Find the common statement ids among hte searched concepts
+        if (statementIdSearchSetList.size() >= 1) {
+            statementIdSet = statementIdSearchSetList.get(0);
+            for (int i = 1; i < statementIdSearchSetList.size(); i++) {
+                statementIdSet = Sets.intersection(statementIdSet, statementIdSearchSetList.get(i));
             }
```

Comment:
I think you can shorten this using Java 8 forEach and retainAll on the set. Also, instead of the if, just use the ternary operator (if statementIdSearchSetList not empty use first one otherwise empty set) and then do the loop without worrying.